libraries/AP_NavEKF2/LogStructure.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <AP_Logger/LogStructure.h>
+#include <AP_AHRS/AP_AHRS.h>
```

Comment:
we should probably move the HAL_NAVEKF2_AVAILABLE out from AP_AHRS.h into a AP_AHRS_Conf.h or something simpler to not push ahrs here and some other place that don't really need it.

But not related to this PR

Reply:
Can we not include all of AHRS.h? That makes touching any header file that's transitively included also rebuild anything that's using logging.... This kind of transitive include path makes incremental builds a lot worse, and actually generally hurts build times.

Reply:
Currently, HAL_NAVEKF2_AVAILABLE is only checked on AHRS.h so that isn't great.
That is why I spoke about a config.h that we could include easily that don't bring the full AHRS that is a pain. Just making a .h in AP_AHRS should be enough to get this

Reply:
Shouldn't `HAL_NAVEKF2_AVAILABLE` be in `AP_NavEKF2.h`?